neuralop/layers/mlp.py

```diff
@@ -44,27 +44,47 @@ def __init__(
             if dropout > 0.0
             else None
         )
-
-        Conv = getattr(nn, f"Conv{n_dim}d")
+        
+        # use nn.Conv1d for everything and roll data along the 1st data dim
         self.fcs = nn.ModuleList()
         for i in range(n_layers):
             if i == 0 and i == (n_layers - 1):
-                self.fcs.append(Conv(self.in_channels, self.out_channels, 1))
+                self.fcs.append(nn.Conv1d(self.in_channels, self.out_channels, 1))
             elif i == 0:
-                self.fcs.append(Conv(self.in_channels, self.hidden_channels, 1))
+                self.fcs.append(nn.Conv1d(self.in_channels, self.hidden_channels, 1))
             elif i == (n_layers - 1):
-                self.fcs.append(Conv(self.hidden_channels, self.out_channels, 1))
+                self.fcs.append(nn.Conv1d(self.hidden_channels, self.out_channels, 1))
             else:
-                self.fcs.append(Conv(self.hidden_channels, self.hidden_channels, 1))
+                self.fcs.append(nn.Conv1d(self.hidden_channels, self.hidden_channels, 1))
 
     def forward(self, x):
+        # if x's data is >1d, reshape into 1d b, c, x
+        reshaped = False
+        size = list(x.shape)
+        if x.ndim > 3:  # batch, channels, x1, x2... extra dims
+            # flatten last data dims
+            if x.is_contiguous():
+                # view only works on contiguous tensors but is better
+                x = x.view(*size[:2], -1) 
+            else:
+                x = x.reshape((*size[:2], -1)) 
```

Comment:
That makes a lot of sense, view is ideal. Here we shouldn’t need it though:
I already rely on the data being in C-order to justify using a reshape + Conv1D. This means that a view should be created under the hood.